cmd/waifu2x/cmd.go

```diff
@@ -56,26 +59,54 @@ func run(opt *option) error {
 		return fmt.Errorf("load noise model, %v", err)
 	}
 
-	model := waifu2x.Waifu2x{
-		Scale2xModel: model2x,
-		NoiseModel:   noise,
-		Scale:        opt.scale,
-		IsDenoising:  true,
+	r, g, b, a, _ := pix.Decompose()
+
+	//XXX ノイズ適用かどうかを決めるようにする
+	if true {
+		r, g, b = noise.Encode(r, g, b)
+	}
+
+	if opt.scale != 1.0 {
+		var err error
```

Comment:
__[gometalinter]__
* [vetshadow] declaration of err shadows declaration at cmd.go:16: